ddtrace/appsec/processor.py

```diff
@@ -27,11 +29,47 @@ def get_rules():
     return get_env("appsec", "rules", default=DEFAULT_RULES)
 
 
+COLLECTED_REQUEST_HEADERS = {
+    "accept",
+    "accept-encoding",
+    "accept-language",
+    "content-encoding",
+    "content-language",
+    "content-length",
+    "content-type",
+    "forwarded",
+    "forwarded-for",
+    "host",
+    "true-client-ip",
+    "user-agent",
+    "via",
+    "x-client-ip",
+    "x-cluster-client-ip",
+    "x-forwarded",
+    "x-forwarded-for",
+    "x-real-ip",
+}
+
+COLLECTED_HEADER_PREFIX = "http.request.headers."
+
+
+def _set_headers(span, headers):
+    for k in headers:
+        if k in COLLECTED_REQUEST_HEADERS:
+            span._set_str_tag(COLLECTED_HEADER_PREFIX + k.lower(), headers[k])
+            continue
+        low = k.lower()
+        if low in COLLECTED_REQUEST_HEADERS:
+            span._set_str_tag(COLLECTED_HEADER_PREFIX + low, headers[k])
+            COLLECTED_REQUEST_HEADERS.add(k)
```

Comment:
why two cases? why adding not-lowered header to the list?
I'm all confused.

Reply:
This was an attempt at following this suggestion https://github.com/DataDog/dd-trace-py/pull/3193#discussion_r800545573 - but it makes things more complex indeed. I will revert I think.
(in this case, the lower version is already in the list)